pkg/keystone/schema.go

```diff
@@ -0,0 +1,35 @@
+package keystone
+
+import (
+	comv1 "github.com/openstack-k8s-operators/keystone-operator/pkg/apis/keystone/v1"
+	util "github.com/openstack-k8s-operators/lib-common/pkg/util"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"io"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"strings"
+)
+
+type schemaOptions struct {
+	DatabaseHostname string
+	SchemaName       string
+	SchemaPassword   string
+}
+
+// SchemaObject func
+func SchemaObject(cr *comv1.KeystoneAPI) unstructured.Unstructured {
+	opts := schemaOptions{cr.Spec.DatabaseHostname, cr.Name, cr.Spec.DatabasePassword}
+
+	decoder := yaml.NewYAMLOrJSONDecoder(strings.NewReader(util.ExecuteTemplateFile("mariadb_schema.yaml", &opts)), 4096)
+	u := unstructured.Unstructured{}
+	err := decoder.Decode(&u)
+	if err == io.EOF {
+		panic(err.Error())
```

Comment:
Maybe, because you could argue that this is unrecoverable and an error in the operator image. My instinct is still probably not, though.

Reply:
sigh, I thought I removed these. I will fix these up no problem